src/lib/rate-limiter/rate-limiter-post.ts

```diff
@@ -27,10 +27,10 @@ const getRateLimiter = (ctx: ExtendedContext): {
 	id: string,
 	rateLimiter: RateLimiterRedis
 } => {
-	if (ctx.state.user?.id) {
+	if (ctx.state.user) {
 		return {
 			type: 'user',
-			id: ctx.state.user.id,
+			id: ctx.state.user.id ?? ctx.state.user.hashedToken ?? '',
```

Comment:
Actually, this is wrong. The anonymous tokens should go to `anonymousRateLimiter`, not to `authenticatedRateLimiter`. @alexey-yarmosh please update in another PR.